tests/docker/test_manifest.py

```diff
@@ -0,0 +1,166 @@
+"""Tests for registry manifest fetcher."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+
+from supervisor.docker.manifest import (
+    DOCKER_HUB,
+    ImageManifest,
+    RegistryManifestFetcher,
+    parse_image_reference,
+)
+
+
+class TestParseImageReference:
```

Comment:
This style of organizing tests is legacy I believe? Any particular reason we're using Test classes here when everything else uses individual test methods?